api/base/views.py

```diff
@@ -1,10 +1,60 @@
 from django.http import JsonResponse
+from django.core.urlresolvers import resolve, reverse
 from rest_framework.decorators import api_view
 from rest_framework.response import Response
+from rest_framework import generics
 
 from api.users.serializers import UserSerializer
 from .utils import absolute_reverse
 
+class JSONAPIBaseView(generics.GenericAPIView):
+
+    permissions_exempt = True
+
+    def _get_embed_partial(self, field_name, field):
+        """Create a partial function to fetch the values of an embedded field. A basic
+        example is to include a Node's children in a single response.
+
+        :param str field_name: Name of field of the view's serializer_class to load
+        results for
+        :return function object -> dict:
+        """
+        def partial(item):
+            embed_value = getattr(item, field.lookup_field, None)
+            view, view_args, view_kwargs = resolve(
+                reverse(
+                    field.view_name,
+                    kwargs={field.lookup_url_kwarg: embed_value}
+                )
+            )
+            view_kwargs.update({
+                'request': self.request,
+                'no_embeds': True
+            })
+            response = view(*view_args, **view_kwargs)
+            return response.data
+        return partial
+
+    def get_serializer_context(self):
+        """Inject request into the serializer context. Additionally, inject partial functions
+        (request, object -> embedd items) if the query string contains embeds, and this
+        is the topmost call to this method (the embed partials call view functions which has
+        the potential to create infinite recursion hence the inclusion of the no_embeds
+        view kwarg to prevent this).
+        """
+        context = super(JSONAPIBaseView, self).get_serializer_context()
+        if self.kwargs.get('no_embeds', False):
```

Comment:
Minor, but false is the default, yes? Could you label it default=False? For a minute I thought you were comparing it to False, and it made no sense.